tests/cases.py

```diff
@@ -257,6 +257,11 @@ def assert_all_data_types_row(
             else:
                 db_mapping[binary_col] = bytes(db_mapping[binary_col])
 
+    # `delta` table format stores `wei` type as string
```

Comment:
Decimals with precision higher than 38 are casted to string. `pyarrow` does not allow downcasting a decimal to a lower precision.